sympy/core/tests/test_power.py

```diff
@@ -630,3 +630,11 @@ def test_powers_of_I():
         1, sqrt(I), I, sqrt(I)**3, -1, -sqrt(I), -I, -sqrt(I)**3,
         1, sqrt(I), I, sqrt(I)**3, -1]
     assert sqrt(I)**(S(9)/2) == -I**(S(1)/4)
+
+
+def test_bin_pow():
+    raises(ValueError, lambda: binpow(3, 1))
+    raises(ValueError, lambda: binpow(3, 1/2))
```

Comment:
Note the difference between `1/2 = 0.5` and `S(1)/2 = Rational(1, 2) = S.Half`. The test will raise anyway, more if the purpose was to test for a Rational number or a float. What about `binpow(3, 2.0)`? Should that work and in that case how?